src/test/java/edu/harvard/iq/dataverse/dataaccess/DataAccessTest.java

```diff
@@ -0,0 +1,78 @@
+package edu.harvard.iq.dataverse.dataaccess;
+
+import edu.harvard.iq.dataverse.DataFile;
+import edu.harvard.iq.dataverse.Dataset;
+import edu.harvard.iq.dataverse.mocks.MocksFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+import java.io.IOException;
+
+
+public class DataAccessTest {
+  DataFile datafile;
+  Dataset dataset;
+  DataAccess dataAccess;
+
+  @BeforeEach
+  void setUp() {
+    datafile = MocksFactory.makeDataFile();
+    dataset = MocksFactory.makeDataset();
+    datafile.setOwner(dataset);
+  }
+
+  @Test
+  void testCreateNewStorageIO_throwsWithoutObject() {
+    assertThrows(IOException.class, () -> {
+      DataAccess.createNewStorageIO(null, "valid-tag");
+    });
+  }
+
+  @Test
+  void testCreateNewStorageIO_throwsWithoutStorageTag() {
+    assertThrows(IOException.class, () -> {
+      DataAccess.createNewStorageIO(datafile, null);
+    });
+  }
+
+  @Test
+  void testCreateNewStorageIO_throwsWithEmptyStorageTag() {
+    assertThrows(IOException.class, () -> {
+      DataAccess.createNewStorageIO(datafile, "");
+    });
+  }
+
+  @Test
+  void testCreateNewStorageIO_throwsOnUnsupported() {
+    assertThrows(IOException.class, () -> {
+      DataAccess.createNewStorageIO(datafile, "valid-tag", "zip-unsupported");
+    });
+  }
+
+  @Test
+  void testCreateNewStorageIO_createsFileAccessIObyDefault() throws IOException {
+    StorageIO<Dataset> storageIo = DataAccess.createNewStorageIO(dataset, "valid-tag");
+    assertTrue(storageIo.getClass().equals(FileAccessIO.class));
+  }
+
+  /*
+  TODO: testing these branches requires the possibility to inject swift and s3 client mocks
```

Comment:
Injecting S3/Swift mocks would definitely be useful, but we shouldn't hold onto these tests as there is no roadmap to add the mocking.

Reply:
Alright, I have removed these!